tools/rna_tools/graphprot/graphprot_train_predict.xml

```diff
@@ -0,0 +1,285 @@
+
+                <section name="prediction_options" title="Prediction options">
+                    <param name="score_thr" type="float" value="0"
+                           label="Set GraphProt average profile peak score threshold for reporting peak regions"
+                           help="Regions with peak score higher or equal the given value are reported (default: 0)"/>
+
+                    <param name="ap_extlr" type="integer" value="5" min="0" max="10"
+                           label="Smoothing parameter for calculating the average profile"
+                           help="Defines the average profile up- and downstream extension to produce the average profile. The mean over small sequence windows (window_length = set_value*2 + 1) is used to get the average profile position-wise scores. A value of 0 means no additional smoothing (== original profile scores), while 10 is applies fairly strong smoothing (default: 5)"/>
+
+                    <param name="max_merge_dist" type="integer" value="0" min="0" max="10"
+                           label="Maximum distance between two peak regions for merging" argument="-merge-dist"
+                           help="By default all non-overlapping regions will be reported. E.g. a distance of 1 means that two regions above the set threshold score will be merged if there is 1 nucleotide that separates the two regions"/>
+
+                    <param name="ws_pred_out" label="Predict whole site instead of profile scores" type="boolean"
+                               truevalue="--ws-pred" falsevalue="" checked="False"
+                               help="Run a whole site prediction instead of calculating profiles (default: false)"/>
+                    <param name="conf_out" label="Output high-confidence (p50) peak regions" type="boolean"
+                               truevalue="--conf-out" falsevalue="" checked="False"
+                               help="Output filtered peak regions BED file or predictions file (if whole site scores prediction enabled), using the median positive training site score for filtering (default: false)"/>
+
+                </section>
+
+            </when>
+
+        </conditional>
+
+    </inputs>
+
+    <outputs>
+
+        <data name="model_out_file" format="txt" from_work_dir="GraphProt.model" label="${tool.name} on ${on_string}: GraphProt model file">
+            <filter>action_type["action_type_selector"] == "train"</filter>
+        </data>
+        <data name="params_out_file" format="txt" from_work_dir="GraphProt.params" label="${tool.name} on ${on_string}: GraphProt model parameters file">
+            <filter>action_type["action_type_selector"] == "train"</filter>
+        </data>
+        <data name="cv_results_out_file" format="txt" from_work_dir="GraphProt.cv_results" label="${tool.name} on ${on_string}: GraphProt cross validation results file">
+            <filter>action_type["action_type_selector"] == "train" and disable_cv is False</filter>
```

Comment:
try:

```suggestion
            <filter>action_type["action_type_selector"] == "train" and not action_type["training_options"]["disable_cv"]</filter>
```

Reply:
Yes, it *is* a problem with the output filters. Sections represent a level of nesting just like conditionals (and repeats), which get expressed as nested dicts in the filters. Adjust the other filters accordingly and things should work.

The `boolean_param is False` vs `not boolean_param` is just cosmetic (the second form is more pythonic).

Reply:
Oh, nice ... it took me too long to find this as well. I'm still not sure of all are working. 

@michauhl can you please add https://github.com/galaxyproject/tools-iuc/blob/91b28adf44f425970469cc52fc5c13e75409ed27/tool_collections/samtools/samtools_markdup/samtools_markdup.xml#L61  to the test to check if the expected output number is correct?